src/code/sys_cfb.c

```diff
@@ -0,0 +1,125 @@
+#include "global.h"
+
+OSViMode sNotebookViMode;
+void* gFramebuffers[2];
+OSViMode* gActiveViMode;
+u16* gZBufferPtr;
+void* gWorkBuffer;
+u64* gGfxSPTaskOutputBufferPtr;
+void* gGfxSPTaskOutputBufferSize; // Actually points to the end of the task buffer
+
+void* sCfbLoRes1;
+void* sCfbLoRes0;
+u16 (*gZBufferLoRes)[SCREEN_WIDTH * SCREEN_HEIGHT];
+u16 (*gWorkBufferLoRes)[SCREEN_WIDTH * SCREEN_HEIGHT];
+u64 (*gGfxSPTaskOutputBufferLoRes)[0x3000];
+void* gGfxSPTaskOutputBufferSizeLoRes; // Actually points to the end of the task buffer
+
+void* sCfbHiRes1;
+void* sCfbHiRes0;
+u16 (*gZBufferHiRes)[SCREEN_WIDTH_NOTEBOOK * SCREEN_HEIGHT_NOTEBOOK];
+u16 (*gWorkBufferHiRes)[SCREEN_WIDTH_NOTEBOOK * SCREEN_HEIGHT_NOTEBOOK];
+u64 (*gGfxSPTaskOutputBufferHiRes)[0x3000];
+void* gGfxSPTaskOutputBufferSizeHiRes; // Actually points to the end of the task buffer
+
+s16 gCfbWidth;
+s16 gCfbHeight;
+s16 gCfbLeftAdjust;
+s16 gCfbUpperAdjust;
+
+u8 gSysCfbHiResEnabled;
+
+void SysCfb_SetLoResMode(void) {
+    gFramebuffers[1] = sCfbLoRes1;
+    gFramebuffers[0] = sCfbLoRes0;
+    gZBufferPtr = *gZBufferLoRes;
+    gWorkBuffer = gWorkBufferLoRes;
+    gGfxSPTaskOutputBufferPtr = *gGfxSPTaskOutputBufferLoRes;
+    gGfxSPTaskOutputBufferSize = gGfxSPTaskOutputBufferSizeLoRes;
+    gCfbWidth = SCREEN_WIDTH;
+    gCfbHeight = SCREEN_HEIGHT;
+    gCfbLeftAdjust = 0;
+    gCfbUpperAdjust = 0;
+    gSysCfbHiResEnabled = false;
+    gScreenWidth = gCfbWidth;
+    gScreenHeight = gCfbHeight;
+    gActiveViMode = &osViModeNtscLan1;
+}
+
+void SysCfb_SetHiResMode(void) {
+    gFramebuffers[1] = sCfbHiRes1;
+    gFramebuffers[0] = sCfbHiRes0;
+    gZBufferPtr = *gZBufferHiRes;
+    gWorkBuffer = gWorkBufferHiRes;
+    gGfxSPTaskOutputBufferPtr = *gGfxSPTaskOutputBufferHiRes;
+    gGfxSPTaskOutputBufferSize = gGfxSPTaskOutputBufferSizeHiRes;
+    if (1) {}
+    gCfbWidth = SCREEN_WIDTH_NOTEBOOK;
+    gCfbHeight = SCREEN_HEIGHT_NOTEBOOK;
+    gCfbLeftAdjust = 30;
+    gCfbUpperAdjust = 10;
+    gScreenWidth = gCfbWidth;
+    gScreenHeight = gCfbHeight;
+    if ((gCfbWidth == SCREEN_WIDTH_HIRES) && (gCfbHeight == SCREEN_HEIGHT_HIRES)) {
+        gActiveViMode = &osViModeNtscHpf1;
+    } else {
+        s32 rightAdjust;
+        s32 lowerAdjust;
+
+    l1:
```

Comment:
```suggestion
```
Is this needed?

Reply:
unfortunately yes. Is there a better name for this?

Reply:
I don't think the name is important, but would be good to add a `//! FAKE:` label. These kind of labels in-function usually indicates there's a scoping issue with the function